lms/services/course.py

```diff
@@ -106,7 +106,7 @@ def find_group_set(self, group_set_id=None, name=None, context_id=None):
         if name:
             query = query.filter(group_set.c.name == name)
 
-        return query.one_or_none()
+        return query.first()
```

Comment:
Not entirely correct, we might just pick one at random but we don't know the unique guaranteess of this until we gather more data in the DB.


Some DB query for future reference:


```
select grouping.application_instance_id, group_sets.id, count(group_sets.id) 
from grouping, jsonb_to_recordset(grouping.extra->'group_sets') as group_sets(id text, name text) 
```

Reply:
Our assumptions now are:

- Getting by ID in the same institution will always get one course
- Getting by name & course will get us the relevant course.

Reply:
Might want to change the docstring to something like `"Find the first matching group set in this course."`